site/source/docs/api_reference/proxying.h.rst

```diff
@@ -0,0 +1,153 @@
+.. _proxying-h:
+
+==========
+proxying.h
+==========
+
+The Emscripten `proxying.h`_
+API provides a mechanism for dispatching work to be executed by a target thread
+and optionally blocking until that work has been completed. This is particularly
+useful in a Web context where certain JS APIs can only be called on particular
+threads; any thread can access those APIs by proxying the API calls to the
+correct threads. Proxying is also useful for wrapping asynchronous JS tasks with
+a synchronous interface. The calling thread can proxy the asynchronous task to a
+dedicated worker thread and wait until the worker thread eventually marks the
+task as complete, possibly after returning multiple times to the JS event loop.
+
+On the target thread, queued work can be executed in two ways. First, the user
+may explicitly call ``emscripten_proxy_execute_queue`` to execute any tasks that
+have been queued for the current thread. Alternatively, queued work will execute
+automatically if the target thread returns to the JS event loop, for example via
+``emscripten_exit_with_live_runtime``.
+
+.. note::
+
+  In situations where the target thread depends on returning to the JS event
+  loop to automatically execute tasks and the proxying of tasks races with the
+  initialization of the target thread, it is possible that the internal
+  notification to execute the queue will arrive before the target thread has
+  been initialized. When that happens the notification is dropped and the queue
```

Comment:
This sounds like a bug, I think - should we file it and eventually fix it? It seems like we could check automatically on thread startup for any queued notifications (I don't believe postMessages can ever actually get lost by the browser, can they?)

Reply:
I agree it would be nice to fix this. I'll file a bug.